File: 501-find-mode-in-binary-search-tree/find-mode-in-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> findMode(TreeNode* root) {
        if (!root) {
            return {};
        }
        
        vector<int> lis;
        
        function<void(TreeNode*)> recur = [&](TreeNode* node) {
            if (!node) return;
            lis.push_back(node->val);
            recur(node->left);
            recur(node->right);
        };
        
        recur(root);
        
        unordered_map<int, int> count;
        for (int val : lis) {
            count[val]++;
        }
        
        int maxFreq = 0;
        for (const auto& pair : count) {
            maxFreq = max(maxFreq, pair.second);
        }
        
        vector<int> result;
        for (const auto& pair : count) {
            if (pair.second == maxFreq) {
                result.push_back(pair.first);
            }
        }
        
        return result;
    }
};

File: 501-find-mode-in-binary-search-tree/find-mode-in-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findMode(self, root: Optional[TreeNode]) -> List[int]:
        if(not root):
            return []
        lis = []
        def recur(node):
            if(not node):
                return
            lis.append(node.val)
            recur(node.left)
            recur(node.right)
        recur(root)
        k = Counter(lis)
        m = 0
        for i in lis:
             m = max(m,k[i])
        res = []
        for j in lis:
            if(k[j]==m):
                res.append(j)
        return list(set(res))
